test/Performance/HelloWorld.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Microsoft.Extensions.PlatformAbstractions;
+using Microsoft.Xunit.Performance;
+using Xunit;
+using System.Diagnostics;
+
+namespace Microsoft.DotNet.Tests.Performance
+{
+    public class HelloWorld : TestBase
+    {
+        private static readonly string NetCoreAppTfm = "netcoreapp1.0";
+        private static readonly string s_expectedOutput = "Hello World!" + Environment.NewLine;
+        private static readonly string s_testdirName = "helloworldtestroot";
+        private static readonly string s_outputdirName = "test space/bin";
+
+        private static string RestoredTestProjectDirectory { get; set; }
+
+        private string Rid { get; set; }
+        private string TestDirectory { get; set; }
+        private string TestProject { get; set; }
+        private string OutputDirectory { get; set; }
+
+        static HelloWorld()
+        {
+            HelloWorld.SetupStaticTestProject();
+        }
+
+        public static void Main()
+        {
+            Console.WriteLine("Dummy Entrypoint.");
+        }
+
+        public HelloWorld()
+        {
+            TestInstanceSetup();
+        }
+
+        [Benchmark]
+        public void MeasureDotNetBuild()
+        {
+            foreach (var iter in Benchmark.Iterations)
+            {
+                // Setup a new instance of the test project.
+                TestInstanceSetup();
+
+                // Setup the build command.
+                var buildCommand = new BuildCommand(TestProject, output: OutputDirectory, framework: NetCoreAppTfm);
+                using (iter.StartMeasurement())
+                {
+                    // Execute the build command.
+                    buildCommand.Execute();
+                }
+            }
+        }
+
+        private void TestInstanceSetup()
+        {
+            var root = Temp.CreateDirectory();
+
+            var testInstanceDir = root.CopyDirectory(RestoredTestProjectDirectory);
```

Comment:
This should use the TestAssetManager. Unfortunately, it's not currently building new directories that are empty. Fortunately, that fix is in https://github.com/dotnet/cli/pull/2653.

This PR, amongst other things, refactors the TestInstance into TestInstance:TestDirectory where the TestDirectory is just an empty directory placed in "the right path" for this test's usage. 

@brianrob could you either wait for that PR or just cherry pick https://github.com/dotnet/cli/pull/2653/commits/1feed70df57558f0777f850e44904b1ca2eb4835 and move the test onto the common infra?